docs/recipes/Mocking-API-Calls.md

```diff
@@ -2,35 +2,121 @@
+
+    ```bash
+    npm run set-middleware [path-to-your-module]/dev.middleware.js
+    ```
+
+   **Using One App runner**
+
+    When using [one-app-runner](https://github.com/americanexpress/one-app-cli/tree/master/packages/one-app-runner), this can be setup within the package.json as follows:
+  
+    ```json
+    "one-amex": {
+      "runner": {
+        "parrotMiddleware": "[path-to-your-module]/dev.middleware.js"
+      },
+    }
+    ```
+
+    The file `dev-middleware.js` contains the following parrot configurations.
+
+    ```javascript
+    const parrot = require('parrot-middleware');
+    const scenarios = require('./mock/scenarios');
+
+    module.exports = (app) => app.use(parrot(scenarios));
+    ```
+
+     - ### `set-dev-endpoints.js`
+
+   A `dev.endpoints.js` file contains all the information One App needs to configure [one-app-dev-proxy](https://github.com/americanexpress/one-app-dev-proxy)
+   (our reverse proxy and mocking server that runs during development) and can be used to set remote endpoints
+   for your Module to use during local development. Create this file within the root folder of your module.
+
+   **Running within One App.**
+
+   Use the `set-dev-endpoints` command to link your module's dev endpoints file to One App. Navigate to your local One App directory and run the following command: 
+
+   ```bash
+    npm run set-dev-endpoints [path-to-your-module]/dev.endpoints.js
+   ```
+
+   **Using One App runner**
+
+   When using [one-app-runner](https://github.com/americanexpress/one-app-cli/tree/master/packages/one-app-runner), this can be setup within the package.json as follows:
+  
+    ```json
+    "one-amex": {
+      "runner": {
+        "devEndpoints": "[path-to-your-module]/dev.endpoints.js"
+      },
+    }
+    ```
+
+   A `dev.endpoints.js` file contains the following:
+
+   ```js
+   module.exports = () => ({
+     apiUrl: {
+       devProxyPath: 'someApiUrl',
+       destination: 'https://example.com',
+     },
+   });
+    ```
+
+1. Start One App using the below command, this will allow the application to start with `one-app-dev-proxy` custom configurations setup above.
+
+    ```bash
+    npm start -- --root-module-name=<module-name> --use-middleware
+    # e.g. npm start -- --root-module-name=my-first-module --use-middleware
+    ```
+
+2. Install the [Parrot Chrome extension](https://chrome.google.com/webstore/detail/parrot-devtools/jckchajdleibnohnphddbiglgpjpbffn) to switch between different scenarios, you can also view the scenarios by visiting [http://localhost:3002/parrot/scenarios](http://localhost:3002/parrot/scenarios)
+
+**📘 More Information**
+
+* [parrot-middleware](https://github.com/americanexpress/one-app-cli/tree/master/packages/one-app-runner#parrot-middleware-optional)
+* [dev-endpoints](https://github.com/americanexpress/one-app-cli/tree/master/packages/one-app-runner#dev-endpoints-optional)
```

Comment:
I would add references to `Loading Data` API documentation so developers can look up how to use `loadModuleData` specifically.

Reply:
agreed, i've added the reference to the docs